We enabled tests for this branch 



![image](https://user-images.githubusercontent.com/4739316/127612000-35b2e47d-1be9-49ec-800d-c9674b488ed0.png)



```
    _issue_warning_captured(deprecated.JUNIT_XML_DEFAULT_FAMILY, config.hook, 2)
/usr/local/lib/python3.8/dist-packages/table_logger/table_logger.py:80
  /usr/local/lib/python3.8/dist-packages/table_logger/table_logger.py:80: DeprecationWarning: `np.float` is a deprecated alias for the builtin `float`.
 To silence this warning, use `float` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type,
 use `np.float64` here.
  Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
    np.float: fmt.FloatFormatter,
/usr/local/lib/python3.8/dist-packages/table_logger/table_logger.py:86
  /usr/local/lib/python3.8/dist-packages/table_logger/table_logger.py:86: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To
silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.i
nt64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
  Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
    np.int: fmt.IntegerFormatter,
-- Docs: https://docs.pytest.org/en/latest/warnings.html
- generated xml file: /mnt/test-data-volume/kubeflow-tf-operator-presubmit-v1-1315-4082e03-8016-e389/output/artifacts/junit_sdk-test.xml -
=========================== short test summary info ============================
FAILED sdk/python/test/test_e2e.py::test_sdk_e2e - RuntimeError: ("Timeout wa...
============= 1 failed, 7 passed, 3 warnings in 602.85s (0:10:02) ==============
Exception ignored in: <function ApiClient.__del__ at 0x7f4972ea8f70>
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/kubernetes/client/api_client.py", line 80, in __del__
  File "/usr/lib/python3.8/multiprocessing/pool.py", line 649, in close
  File "/usr/lib/python3.8/multiprocessing/queues.py", line 362, in put
AttributeError: 'NoneType' object has no attribute 'dumps'
```

```
INFO|2021-07-30T06:41:25|/mnt/test-data-volume/kubeflow-tf-operator-presubmit-v1-1315-4082e03-8016-e389/src/kubeflow/tf-operator/py/kubeflow/tf_operato
r/tf_job_client.py|109| Job simple-tfjob-tests-v1 in namespace kubeflow; uid=171ed79f-1cb5-4680-9dd3-4eb790106bf3; conditions=[]
INFO|2021-07-30T06:41:55|/mnt/test-data-volume/kubeflow-tf-operator-presubmit-v1-1315-4082e03-8016-e389/src/kubeflow/tf-operator/py/kubeflow/tf_operato
r/tf_job_client.py|109| Job simple-tfjob-tests-v1 in namespace kubeflow; uid=171ed79f-1cb5-4680-9dd3-4eb790106bf3; conditions=[]
INFO|2021-07-30T06:42:25|/mnt/test-data-volume/kubeflow-tf-operator-presubmit-v1-1315-4082e03-8016-e389/src/kubeflow/tf-operator/py/kubeflow/tf_operato
r/tf_job_client.py|109| Job simple-tfjob-tests-v1 in namespace kubeflow; uid=171ed79f-1cb5-4680-9dd3-4eb790106bf3; conditions=[]
```

SDK tests failed and

--- a/cmd/tf-operator.v1/app/options/options.go
+++ b/cmd/tf-operator.v1/app/options/options.go
@@ -51,7 +51,7 @@ func NewServerOption() *ServerOption {
 
 // AddFlags adds flags for a specific CMServer to the specified FlagSet.
 func (s *ServerOption) AddFlags(fs *flag.FlagSet) {
-	fs.StringVar(&s.Kubeconfig, "kubeconfig", "", "The path of kubeconfig file")
+	//fs.StringVar(&s.Kubeconfig, "kubeconfig", "", "The path of kubeconfig file")
 
 	fs.StringVar(&s.MasterURL, "master", "",
 		`The url of the Kubernetes API server,

--- a/pkg/controller.v1/mxnet/mxjob_controller.go
+++ b/pkg/controller.v1/mxnet/mxjob_controller.go
@@ -422,7 +422,7 @@ func (r *MXJobReconciler) UpdateJobStatusInApiServer(job interface{}, jobStatus 
 		mxJob.Status = *jobStatus.DeepCopy()
 	}
 
-	if err := r.Update(context.Background(), mxJob); err != nil {
+	if err := r.Status().Update(context.Background(), mxJob); err != nil {
 		logrus.Error(err, " failed to update MxJob conditions in the API server")
 		return err
 	}

--- a/pkg/controller.v1/pytorch/pytorchjob_controller.go
+++ b/pkg/controller.v1/pytorch/pytorchjob_controller.go
@@ -406,7 +406,7 @@ func (r *PyTorchJobReconciler) UpdateJobStatusInApiServer(job interface{}, jobSt
 		pytorchjob.Status = *jobStatus.DeepCopy()
 	}
 
-	result := r.Update(context.Background(), pytorchjob)
+	result := r.Status().Update(context.Background(), pytorchjob)
 
 	if result != nil {
 		r.Log.WithValues("pytorchjob", types.NamespacedName{

--- a/pkg/controller.v1/tensorflow/job.go
+++ b/pkg/controller.v1/tensorflow/job.go
@@ -112,6 +112,8 @@ func (tc *TFController) addTFJob(obj interface{}) {
 	}
 
 	// Set default for the new tfjob.
+	// TODO(Jeffwan): Consider to change to scheme https://github.com/kubeflow/tf-operator/issues/1317#issuecomment-890397705
+	tfv1.SetDefaults_TFJob(tfJob)
 	scheme.Scheme.Default(tfJob)
 
 	msg := fmt.Sprintf("TFJob %s is created.", tfJob.Name)

--- a/pkg/controller.v1/tensorflow/tfjob_controller.go
+++ b/pkg/controller.v1/tensorflow/tfjob_controller.go
@@ -524,7 +524,7 @@ func (r *TFJobReconciler) UpdateJobStatusInApiServer(job interface{}, jobStatus 
 	tfJob = tfJob.DeepCopy()
 	tfJob.Status = *jobStatus.DeepCopy()
 
-	result := r.Update(context.Background(), tfJob)
+	result := r.Status().Update(context.Background(), tfJob)
 
 	if result != nil {
 		r.Log.WithValues("tfjob", types.NamespacedName{

--- a/pkg/controller.v1/xgboost/xgboostjob_controller.go
+++ b/pkg/controller.v1/xgboost/xgboostjob_controller.go
@@ -415,7 +415,7 @@ func (r *XGBoostJobReconciler) UpdateJobStatusInApiServer(job interface{}, jobSt
 		xgboostjob.Status.JobStatus = *jobStatus.DeepCopy()
 	}
 
-	result := r.Update(context.Background(), xgboostjob)
+	result := r.Status().Update(context.Background(), xgboostjob)
 
 	if result != nil {
 		logger.LoggerForJob(xgboostjob).Error(result, "failed to update XGBoost Job conditions in the API server")

